homeassistant/components/sensor/onewire.py

```diff
@@ -0,0 +1,93 @@
+""" Support for DS18B20 One Wire Sensors"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELCIUS, TEMP_FAHRENHEIT
+from glob import glob
+import os
+import time
+import logging
+
+
+BASE_DIR = '/sys/bus/w1/devices/'
+DEVICE_FOLDERS = glob(os.path.join(BASE_DIR, '28*'))
+SENSOR_IDS = [os.path.split(device_folder)[1] for device_folder in DEVICE_FOLDERS]
+DEVICE_FILES = [os.path.join(device_folder, 'w1_slave') for device_folder in DEVICE_FOLDERS]
+
+_LOGGER = logging.getLogger(__name__)
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the one wire Sensors"""
+    # TODO check if kernel modules are loaded
```

Comment:
As far as I know, the files do not show up, but I have not tested it. My
Idea was to give the user a hint, why it is not working, if the sensors
can not be loaded.

On 29.03.2015 17:50, Paulus Schoutsen wrote:
>> +import os
>> +import time
>> +import logging
>> +
>> +
>> +BASE_DIR = '/sys/bus/w1/devices/'
>> +DEVICE_FOLDERS = glob(os.path.join(BASE_DIR, '28*'))
>> +SENSOR_IDS = [os.path.split(device_folder)[1] for device_folder in DEVICE_FOLDERS]
>> +DEVICE_FILES = [os.path.join(device_folder, 'w1_slave') for device_folder in DEVICE_FOLDERS]
>> +
>> +_LOGGER = logging.getLogger(__name__)
>> +
>> +# pylint: disable=unused-argument
>> +def setup_platform(hass, config, add_devices, discovery_info=None):
>> +    """ Sets up the one wire Sensors"""
>> +    # TODO check if kernel modules are loaded
> If the kernel module is not loaded, will the files still show up in `/sys/bus/w1/devices` folder?
>
> The style tests do not allow TODO to be added to the codebase. If it is an idea and not something that still has to be done to get it working, replace it with FUTURE.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/balloob/home-assistant/pull/78/files#r27355580